packages/flame_riverpod/lib/src/component_ref.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+/// A wrapper around [WidgetRef]. Methods that correspond to interactions with
+/// the widget tree, such as [WidgetRef.watch] are not exposed.
```

Comment:
Is there a way to explain what this class is without referring to `WidgetRef`? And what does the name mean, "component reference"?

Reply:
The name Component Reference exists only to suggest that it is distinct in some sense from a Widget Reference - that the APIs between the two may differ, and to prompt a consuming user to investigate further. 

Specifically, at this time, **watch**, the typically recommended API for subscribing to a value exposed by a provider in Riverpod is not exposed directly. The reason for this is to prevent unnecessary rebuilds